pymc/tests/test_model.py

```diff
@@ -702,6 +703,64 @@ def test_nested_model_coords():
     assert set(m2.RV_dims) < set(m1.RV_dims)
 
 
+def test_shape_error_from_fixed_dims():
+    """
+    with #5751 dimensions set by add_coord() are default TensorConstant type
+    and so have fixed dimensions. Trying to resize them will raise a ShapeError,
+    which this test checks for.
+    """
+    with pm.Model() as pmodel:
+        a = pm.Normal("a", mu=[1, 2, 3], dims="fixed")
+
+        m = pm.MutableData("m", [[1, 2, 3]], dims=("one", "fixed"))
+
+        # This is fine because the "fixed" dim is not resized
+        pm.set_data({"m": [[1, 2, 3], [3, 4, 5]]})
+
+    with pytest.raises(ShapeError, match="was initialized from 'a'"):
+        # Can't work because the "fixed" dimension is linked to a constant shape:
+        # Note that the new data tries to change both dimensions
+        with pmodel:
+            pm.set_data({"m": [[1, 2], [3, 4]]})
+
+
+def test_value_error_from_non_fixed_dims_no_coords():
+    """
+    test that resizing dims without passing coords raises a ValueError.
+    """
+    with pm.Model() as pmodel:
+        pmodel.add_coord("shared", [1, 2, 3], fixed=False)
+        m = pm.MutableData("m", [1, 2, 3], dims=("shared"))
+        with pytest.raises(ValueError, match="'m' variable already had 3"):
+            # tries to resize m but without passing coords so raise ValueError
+            pm.set_data({"m": [1, 2, 3, 4]})
+
+
+def test_add_coord_fixed_dims():
+    """
+    test that dims does not cause resizing of y. See #5181.
+    """
+    with pm.Model(coords={"feature": [1], "group": ["A", "B"]}):
+        x = pm.Normal("x", 0, 1, dims="feature")
+        y = pm.Normal("y", x, 1, dims=("group", "feature"))
+    assert x.eval().shape == (1,)
+    assert y.eval().shape == (2, 1)
```

Comment:
```suggestion
def test_coords_and_constantdata_create_fixed_dims():
    """
    When created from `pm.Model(coords=...)` or `pm.ConstantData`
    a dimension should be fixed in length.
    """
    with pm.Model(coords={"group": ["A", "B"]}) as m:
        x = pm.ConstantData("x", [0], dims="feature")
        y = pm.Normal("y", x, 1, dims=("group", "feature"))
    assert isinstance(m._dim_lengths["feature"], TensorConstant)
    assert isinstance(m._dim_lengths["group"], TensorConstant)
    assert x.eval().shape == (1,)
    assert y.eval().shape == (2, 1)
```

Reply:
Why does the assert statement on line 759 fail? Its created from coord so should be a `TensorConstant` but is `Subtensor{int64}.0`.